eng/pipelines/maui-release.yml

```
@@ -190,4 +196,85 @@ extends:
+        - ${{ if eq(variables['Build.Reason'], 'Schedule')) }}:
+          - stage: nightly
+            displayName: Nightly
+            dependsOn: ['sdk_insertion']
+            jobs:
+            - job: push_nightly
+              displayName: Push to nightly feed
+              pool: ${{ parameters.VM_IMAGE_HOST }}
+              workspace:
+                clean: all
+              timeoutInMinutes: 60
+              steps:
+              - checkout: none
+
+              - task: UseDotNet@2
+                displayName: Use .NET 8.x
+                inputs:
+                  version: 8.x
+
+              - task: AzureCLI@2
+                displayName: Update service connection with managed identity
+                inputs:
+                  azureSubscription: maui-nightly-feed-mi
+                  scriptType: ps
+                  scriptLocation: inlineScript
+                  inlineScript: |
+                    $accessToken = az account get-access-token --query accessToken --resource $(MauiNightlyFeedMIResourceId) -o tsv
+                    Write-Host "##vso[task.setsecret]$accessToken"
+                    Write-Host "Overwriting authorization token for the maui-nightly-feed service connection"
+                    # https://learn.microsoft.com/en-us/azure/devops/pipelines/scripts/logging-commands?view=azure-devops&tabs=bash#setendpoint-modify-a-service-connection-field
+                    Write-Host "##vso[task.setendpoint id=$(MauiNightlyFeedServiceConnectionId);field=authParameter;key=apitoken]${accessToken}"
+
+              - task: NuGetAuthenticate@0
+                displayName: NuGet Authenticate using managed identity token
+                inputs:
+                  nuGetServiceConnections: maui-nightly-feed
+
+              - task: DownloadPipelineArtifact@2
+                inputs:
+                  artifactName: DropMetadata-$(Build.BuildId)-nugets-$(System.JobAttempt)
+                  downloadPath: $(Build.StagingDirectory)\metadata
+                displayName: Download nugets drop metadata
+
+              - powershell: |
+                  $jsonContent = Get-Content -Path "$(Build.StagingDirectory)\metadata\VSTSDrop.json" -Raw | ConvertFrom-Json
+                  $dropPrefix = $jsonContent.VstsDropBuildArtifact.VstsDropUrl -replace 'https://devdiv.artifacts.visualstudio.com/DefaultCollection/_apis/drop/drops/' -replace '/nugets'
+                  Write-Host "##vso[task.setvariable variable=ReleaseDropPrefix]$dropPrefix"
+                displayName: Set variable ReleaseDropPrefix
+
+              - task: ms-vscs-artifact.build-tasks.artifactDropDownloadTask-1.artifactDropDownloadTask@1
+                displayName: Download $(ReleaseDropPrefix)/nugets
+                inputs:
+                  dropServiceURI: https://devdiv.artifacts.visualstudio.com/DefaultCollection
+                  buildNumber: $(ReleaseDropPrefix)/nugets
+                  destinationPath: $(Build.StagingDirectory)\nuget-signed
```

This currently relies on https://github.com/dotnet/maui/pull/23658, but should probably be updated to download from `nuget-signed` and `vs-msi-nugets` artifacts instead, assuming we only want to use this to .NET 8 builds from main for now.

```suggestion
              - task: DownloadPipelineArtifact@2
                inputs:
                  artifactName: nuget-signed
                  downloadPath: $(Build.StagingDirectory)\nuget-signed
                displayName: Download nuget-signed

             - task: DownloadPipelineArtifact@2
                inputs:
                  artifactName: vs-msi-nugets
                  downloadPath: $(Build.StagingDirectory)\nuget-signed
                displayName: Download vs-msi-nugets
```